xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -333,25 +333,15 @@ void CGUIDialogAddonInfo::OnInstall()
     return;
 
   std::string addonId = m_item->GetAddonInfo()->ID();
-  std::vector<std::pair<AddonVersion, std::string>> versions;
+  AddonVersion version = m_item->GetAddonInfo()->Version();
+  std::string origin = m_item->GetAddonInfo()->Origin();
 
-  CAddonDatabase database;
-  if (!database.Open() || !database.GetAvailableVersions(addonId, versions) || versions.empty())
-  {
-    CLog::Log(LOGERROR, "ADDON: no available versions of %s", addonId.c_str());
+  Close();
+  auto deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(addonId);
+  if (!deps.empty() && !ShowDependencyList(deps, false))
     return;
-  }
 
-  int i = versions.size() == 1 ? 0 : AskForVersion(versions);
-  if (i != -1)
-  {
-    Close();
-    auto deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(m_item->GetAddonInfo()->ID());
-    if (!deps.empty() && !ShowDependencyList(deps, false))
-      return;
-
-    CAddonInstaller::GetInstance().Install(addonId, versions[i].first, versions[i].second);
-  }
```

Comment:
About here I'm must test to see that still somewhere older versions possible to select/use.

Know also for me (specially by binary addons) that it can good to allow select of older version.

Reply:
@AlwinEsch once you install an add-on you can go the AddonInfoDialog and click `Update` if you want to install an older version from that repo. We just won't allow that from the `Install` button.